### PR TITLE
feat: makes space for `EncryptedPublicationMetadataUploader`

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/api-bindings/src/metadata.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  traitType?: string;
+  value: string;
+};
+
+export type ProfileMetadata = {
+  version: '1.0.0';
+  metadata_id: string;
+  attributes: ProfileMetadataAttribute[];
+  bio: string | null;
+  cover_picture: string | null;
```
Comment (desfero) [inline packages/api-bindings/src/metadata.ts:21]: nit: unusual to see pascal case here over the camelcase.

File: packages/react/src/transactions/infrastructure/PublicationMetadataUploader.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { createPublicationMetadata } from './createPublicationMetadata';
+
+export class PublicationMetadataUploader<T extends CreatePostRequest | CreateCommentRequest>
+  implements IMetadataUploader<T>
+{
+  constructor(private readonly handler: MetadataUploadHandler) {}
+
+  async upload(request: T): Promise<Url> {
+    try {
+      const metadata = createPublicationMetadata(request);
```
Comment (desfero) [inline packages/react/src/transactions/infrastructure/PublicationMetadataUploader.ts:18]: nit: just API disproportion which can be good,  `PublicationMetadataUploader` takes the `request` whereas the `ProfileMetadataUploader` takes the metadata directly.

File: packages/react/src/transactions/useCreatePost.ts [hunk 3]
```diff
@@ -33,7 +36,8 @@ export type CreatePostOperation = Operation<
 >;
 
 export function useCreatePost({ publisher, upload }: UseCreatePostArgs): CreatePostOperation {
-  const createPost = useCreatePostController({ upload });
+  const uploader = useMemo(() => new PublicationMetadataUploader(upload), [upload]);
```
Comment (desfero) [inline packages/react/src/transactions/useCreatePost.ts:39]: cool, no changes to the consumer-facing API

**Review by desfero — APPROVED** [review]

File: packages/api-bindings/src/metadata.ts (hunk 1 above)
Comment (cesarenaldi) [reply packages/api-bindings/src/metadata.ts:21]: This is the metadata JSON file format, we have to use what the metadata specification tells u.

File: packages/react/src/transactions/infrastructure/PublicationMetadataUploader.ts (hunk 2 above)
Comment (cesarenaldi) [reply packages/react/src/transactions/infrastructure/PublicationMetadataUploader.ts:18]: I know, that's why `T` in `IMetadataUploader<T>` is without restrictions.

**Pull request merged**